packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -1,14 +1,22 @@
-import { GraphQLClient } from 'graphql-request';
-import chalk from 'chalk';
+import { GraphQLClient, ClientError } from 'graphql-request';
 import { DocumentNode } from 'graphql';
+import { debugHttp as debug } from './debug';
 
 export class GraphQLRequestClient {
+  private client: GraphQLClient;
+  private headers: Record<string, string> = {};
+
   /**
    * Provides ability to execute graphql query using given `endpoint`
    * @param {string} endpoint The Graphql endpoint
    * @param {string} [apiKey] The API key to use for authentication. This will be added as an 'sc_apikey' header.
    */
-  constructor(private endpoint: string, private apiKey?: string) {}
+  constructor(private endpoint: string, apiKey?: string) {
```

Comment:
The api key needs to be exposed publicly?

Reply:
It's no longer a property at all - instead used to populate the private `headers` property in the constructor body.